props/saber_fett263_buttons.h

```diff
@@ -293,248 +660,3355 @@ class SaberFett263Buttons : public virtual PropBase {
+      case COLOR_LOCKUP:
+      case COLOR_LB:
+        menu_type_ = MENU_COLOR_MODE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        NewColor(blade_num_, effect_num_); 
+        current_preset_.Save();
+        bump_color_.Stop(blade_num_);
+        UpdateStyle(current_preset_.preset_num);
+        MenuSave();
+        color_mode_ = NONE;
+        break;
+      case COLOR_DRAG:
+      case COLOR_STAB:
+        menu_type_ = MENU_COLOR_MODE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        NewColor(blade_num_, effect_num_); 
+        current_preset_.Save();
+        tip_color_.Stop(blade_num_);
+        UpdateStyle(current_preset_.preset_num);
+        MenuSave();
+        color_mode_ = NONE;
+        break;
+      case COLOR_PREON:
+      case COLOR_PSTOFF:
+        menu_type_ = MENU_COLOR_MODE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        NewColor(blade_num_, effect_num_); 
+        current_preset_.Save();
+        pre_color_.Stop(blade_num_);
+        UpdateStyle(current_preset_.preset_num);
+        MenuSave();
+        color_mode_ = NONE;
+        break;
+      case COLOR_EMITTER:
+        menu_type_ = MENU_COLOR_MODE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        NewColor(blade_num_, effect_num_); 
+        current_preset_.Save();
+        hilt_color_.Stop(blade_num_);
+        UpdateStyle(current_preset_.preset_num);
+        MenuSave();
+        color_mode_ = NONE;
+        break;
+      case MENU_FONT:
+        restore_point = nullptr;
+        menu_type_ = MENU_TOP;
+        MenuSave();
+        break;
+      case MENU_TRACK:
+        restore_point = nullptr;
+        if (track_player_) {
+          track_player_->Stop();
+          track_player_.Free();
+        }
+        menu_type_ = MENU_TOP;
+        MenuSave();
+        break;
+      case MENU_LENGTH:
+        menu_type_ = MENU_SETTING_SUB;
+        current_preset_.Save();
+        show_length_.Stop(blade_num_);
+        UpdateStyle(current_preset_.preset_num);
+        MenuSave();
+        break;
+      case MENU_COPY:          
+        if (choice_) {
+          int32_t pos = current_preset_.preset_num;
+          current_preset_.preset_num = -1;
+          current_preset_.SaveAt(pos);
+          menu_type_ = MENU_TOP;
+          MenuSelect();
+          choice_ = false;
+          break;
+        }
+        PlayMenuSound("mconfirm.wav");
+        break;
+      case MENU_RESET_COLOR:
+        if (choice_) {
+          SaberBase::SetVariation(0);
+          current_preset_.SetStyle(blade_num_,  style_parser.ResetArguments(current_preset_.GetStyle(blade_num_)));
+          current_preset_.Save();
+          UpdateStyle(current_preset_.preset_num);
+          menu_type_ = MENU_COLOR_SUB;
+          MenuSelect();
+          choice_ = false;
+          break;
+        }
+        PlayMenuSound("mconfirm.wav");
+        break;
+      case MENU_GESTURE_SUB:
+        switch (effect_num_) {
```

Comment:
Here you seem to be using effect_num_ for something different than above.
That is going to be confusing, and when we turn the values in this function into enums, it's going to be incompatible.
Please create a new variable for this menu, make it an enum and replace the number below with enum labels.

Reply:
replaced with gesture_num_, added enum